File: app/src/main/java/ru/gsench/githubusers/domain/interactor/UserModel.java
package ru.gsench.githubusers.domain.interactor;

import ru.gsench.githubusers.domain.github_repo.GitHubUserShort;

/**
 * Created by grish on 11.05.2017.
 */

public class UserModel extends GitHubUserShort {

    public boolean isFavorite() {
        return isFavorite;
    }

    private boolean isFavorite = false;

    public UserModel(int id, String login) {
        super(id, login);
    }

    public UserModel(int id, String login, boolean favorite) {
        super(id, login);
        isFavorite = favorite;
    }

    public UserModel(GitHubUserShort user, boolean favorite){
        super(user.getId(), user.getLogin());
        this.isFavorite=favorite;
    }

    public UserModel setFavorite(boolean favor){
        isFavorite=favor;
        return this;
    }
}

File: app/src/main/java/ru/gsench/githubusers/domain/interactor/UserModel.kt
package ru.gsench.githubusers.domain.interactor

import ru.gsench.githubusers.domain.github_repo.GitHubUserShort

/**
 * Created by grish on 11.05.2017.
 */

class UserModel : GitHubUserShort {

    fun isFavorite(): Boolean {
        return isFavorite
    }

    private var isFavorite = false

    constructor(id: Int, login: String) : super(id, login) {}

    constructor(id: Int, login: String, favorite: Boolean) : super(id, login) {
        isFavorite = favorite
    }

    constructor(user: GitHubUserShort, favorite: Boolean) : super(user.id, user.login) {
        this.isFavorite = favorite
    }

    fun setFavorite(favor: Boolean): UserModel {
        isFavorite = favor
        return this
    }
}
